src/redux/books/books.js

```diff
@@ -1,20 +1,59 @@
+import BookDataService from '../../services/BookService';
+
+const RETRIEVE_BOOKS = 'bookstore/books/RETRIEVE_BOOKS';
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
-const initBooks = [{ title: 'book1', author: 'author1', id: '1' }, { title: 'book2', author: 'author2', id: '2' }];
+export const retrieveBooks = () => async (dispatch) => {
+  try {
+    const res = await BookDataService.getAll();
+    const { data } = res;
+    const books = Object.keys(data).map((key) => ({
+      id: key,
+      ...data[key][0],
+    }));
+    dispatch({
+      type: RETRIEVE_BOOKS,
+      books,
+    });
+  } catch (err) {
+    console.log(err);
+  }
+};
 
-export function addBook(book) {
-  return { type: ADD_BOOK, book };
-}
+export const addBook = (book) => async (dispatch) => {
+  try {
+    const res = await BookDataService.create(book);
+    dispatch({
+      type: ADD_BOOK,
+      book,
+    });
+    return Promise.resolve(res.data);
+  } catch (err) {
+    return Promise.reject(err);
+  }
+};
 
-export function removeBook(id) {
-  return { type: REMOVE_BOOK, id };
-}
+export const removeBook = (id) => async (dispatch) => {
+  try {
+    await BookDataService.remove(id);
+    dispatch({
+      type: REMOVE_BOOK,
+      id,
+    });
+  } catch (err) {
+    console.log(err);
+  }
+};
```

Comment:
- [ ] Nice work implementing all project's requirements. I noticed that when I add a book and click the `remove` button, the book section disappears. When I reload the page, I see all the books including the one I intended to delete. When I reload the page a second time and click the remove button, I noticed that the changes persist in the server. Kindly check your method for deleting a book and fix the bug. 👍